shared/reports/totals.py

```diff
@@ -0,0 +1,59 @@
+from typing import Iterator
+
+from shared.helpers.numeric import ratio
+from shared.reports.types import ReportLine, ReportTotals
+from shared.utils.merge import LineType, line_type
+
+
+def get_line_totals(lines: Iterator[ReportLine]) -> ReportTotals:
+    """
+    Calculates the totals (`ReportTotals`) across all the given `lines` (`ReportLine`s).
+    """
+    hits = 0
+    misses = 0
+    partials = 0
+    branches = 0
+    methods = 0
+    messages = 0
+    complexity = 0
+    complexity_total = 0
+
+    for line in lines:
+        match line_type(line.coverage):
```

Comment:
[nit] `line_type` and `line.type` (below) are confusing... Maybe we can rename `line_type(` something like `line_coverage_type` or `line_coverage`

If they are the same thing, why can't we match the block below in this `match`?
```
if line.type == "b":
    branches += 1
elif line.type == "m":
    methods += 1
```

Reply:
I renamed the imports for the `line_type` and `LineType` above to make the distinction a little clearer.